packages/near-lake-primitives/src/types/events.ts

```diff
@@ -0,0 +1,30 @@
+
+export class Event {
+    constructor(readonly relatedReceiptId: string, readonly rawEvent: RawEvent) { }
+
+    static fromLog = (log: string): Event | undefined => {
+        const rawEvent = RawEvent.fromLog(log);
+        if (rawEvent) {
+            return new Event('', rawEvent);
+        }
+        return
+    }
+}
+
+export class RawEvent {
+    constructor(readonly event: string, readonly standard: string, readonly version: string, readonly data: JSON | undefined) { }
+
+    static fromLog = (log: string): RawEvent | undefined => {
+        if (!log.startsWith('EVENT_JSON:')) {
+            return
```

Comment:
```suggestion
    static fromLog = (log: string): RawEvent => {
        if (!log.startsWith('EVENT_JSON:')) {
            throw new Error('Log is not and event');
```
Ideally we should throw here. It's seems odd for a `static` method which is used to create the `class` would return nothing, we should instead notify the caller of why we can't create the class.

Reply:
Good point. My concern is in how this function is used. 

```
 const events = this.receipts().flatMap((executedReceipt) => executedReceipt.logs.map(RawEvent.fromLog).map((rawEvent) => {
            if (rawEvent) {
                let event: Event = { relatedReceiptId: executedReceipt.receiptId, rawEvent: rawEvent }
                return event
            } else {
                return null
            }
```

As you can see that we are mapping logs into RawEvents and not all logs will be valid events. In this case if we throw an error it will stop the execution of any indexer. Perhaps we should filter by `RawEvent.isEvent` static method and then we can map them.